File: solutions/java/gigasecond/1/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

public class Gigasecond {

    private static final int GIGASECOND = 1000000000;

    private final LocalDateTime dateTime;

    public Gigasecond(LocalDate moment) {
        dateTime = moment.atStartOfDay().plusSeconds(GIGASECOND);
    }

    public Gigasecond(LocalDateTime moment) {
        dateTime = moment.plusSeconds(GIGASECOND);
    }

    public LocalDateTime getDateTime() {
        return dateTime;
    }
}

File: solutions/kotlin/gigasecond/1/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime

class Gigasecond {
    private val GIGASECOND_SECONDS: Long = 1_000_000_000
    val date: LocalDateTime

    // Constructor Overload
    constructor(localDate: LocalDate) {
        this.date = localDate.atStartOfDay().plusSeconds(GIGASECOND_SECONDS)
    }

    // Constructor Overload
    constructor(localDateTime: LocalDateTime) {
        this.date = localDateTime.plusSeconds(GIGASECOND_SECONDS)
    }
}